src/prefect/engine/cloud/flow_runner.py

```diff
@@ -123,10 +126,104 @@ def call_runner_target_handlers(self, old_state: State, new_state: State) -> Sta
             )
             raise ENDRUN(state=new_state)
 
+        if state.is_queued():
+            state.state = old_state
+            raise ENDRUN(state=state)
+
         prefect.context.update(flow_run_version=version + 1)
 
         return new_state
 
+    def run(
+        self,
+        state: State = None,
+        task_states: Dict[Task, State] = None,
+        return_tasks: Iterable[Task] = None,
+        parameters: Dict[str, Any] = None,
+        task_runner_state_handlers: Iterable[Callable] = None,
+        executor: "prefect.engine.executors.Executor" = None,
+        context: Dict[str, Any] = None,
+        task_contexts: Dict[Task, Dict[str, Any]] = None,
+    ) -> State:
+        """
+        The main endpoint for FlowRunners.  Calling this method will perform all
+        computations contained within the Flow and return the final state of the Flow.
+
+        Args:
+            - state (State, optional): starting state for the Flow. Defaults to
+                `Pending`
+            - task_states (dict, optional): dictionary of task states to begin
+                computation with, with keys being Tasks and values their corresponding state
+            - return_tasks ([Task], optional): list of Tasks to include in the
+                final returned Flow state. Defaults to `None`
+            - parameters (dict, optional): dictionary of any needed Parameter
+                values, with keys being strings representing Parameter names and values being
+                their corresponding values
+            - task_runner_state_handlers (Iterable[Callable], optional): A list of state change
+                handlers that will be provided to the task_runner, and called whenever a task changes
+                state.
+            - executor (Executor, optional): executor to use when performing
+                computation; defaults to the executor specified in your prefect configuration
+            - context (Dict[str, Any], optional): prefect.Context to use for execution
+                to use for each Task run
+            - task_contexts (Dict[Task, Dict[str, Any]], optional): contexts that will be provided to each task
+
+        Returns:
+            - State: `State` representing the final post-run state of the `Flow`.
+        """
+        context = context or {}
+        # Pre-emptively load the flow run id that exists in the global
+        # context into the run specific context, which is needed
+        # specifically for entering a Queued state and waiting forever until
+        # an exit condition occurs. Prefect's context will have the flow_run_id
+        # on the first call to run, and `context` will have it on all subsequent runs
+        if prefect.context.get("flow_run_id") and not context.get("flow_run_id"):
```

Comment:
This feels hacky to me. Is there any way to persist the `flow_run_id` in the `prefect.context` across multiple `run` calls? After light testing, it appears to reset after each call.